### DISCUSSION TITLE
change proposal - samples: net: dhcpv4_client: added disconnect signaling & dhcp restart upon disconnect

### DISCUSSION THREAD

**Discussion by rjmaris** [zephyrproject-rtos/zephyr#75377] [RFC]
I have used the dhcp client sample code as a blueprint for using in an application that offers a system info page showing IP address and gateway.  In the target application I realized simple code extensions to flag disconnect and let the dhcp-client restart after this event.
Now, this handling is transferred to my local sample code. I think it's helpful for others who can see in this code how to do that - see attachment.
[0001-samples-net-hcpv4_client-added-signaling-of-disconne.patch.txt](https://github.com/user-attachments/files/16082872/0001-samples-net-hcpv4_client-added-signaling-of-disconne.patch.txt)

**Comment by jukkar** [comment]
For DHCPv4 events, there is also the `NET_EVENT_IPV4_DHCP_BOUND` event that is fired when one gets IPv4 address from the server.
Not sure why you would restart DHCPv4 if the address is deleted, that can only happen if one manually deletes the address but why that would happen.

**Comment by rjmaris** [comment]
Without the restart invocation, it can take approx. 40 seconds after reconnecting the cable, before the new connection is negotiated.
That's why I did so.
With limited knowledge in details I cannot judge whether a NET_EVENT_IPV4_DHCP_BOUND usage would constitute an alternative way to handle/flag disconnects and reconnects.
Whether way it's done: an extension of the sample code makes sense. ;-)

**Reply by jukkar** [reply]
> With limited knowledge in details I cannot judge whether a NET_EVENT_IPV4_DHCP_BOUND usage would constitute an alternative way to handle/flag disconnects and reconnects.

The bound event would not help here for your use case.
